pkg/backend/httpstate/stack.go

```diff
@@ -230,6 +234,22 @@ func (s *cloudStack) Snapshot(ctx context.Context, secretsProvider secrets.Provi
 	return snap, nil
 }
 
+func (s *cloudStack) SnapshotStackOutputs(
+	ctx context.Context, secretsProvider secrets.Provider,
+) (resource.PropertyMap, error) {
+	if v := s.snapshotStackOutputs.Load(); v != nil {
+		return *v, nil
+	}
+
+	outputs, err := s.b.getSnapshotStackOutputs(ctx, secretsProvider, s.ref)
+	if err != nil {
+		return nil, err
+	}
+
+	s.snapshotStackOutputs.Store(&outputs)
+	return outputs, nil
```

Comment:
if two calls to this method race, they could return inconsistent results. is that acceptable? or should we use `CompareAndSwap(nil, &outputs)`? e.g.

```go
s.snapshotStackOutputs.CompareAndSwap(nil, &outputs)
return *s.snapshotStackOutputs.Load(), nil
```